src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -1019,13 +1089,30 @@ class SceneryWindow final : public Window
         _tabSelections[tabIndex] = value;
     }
 
-    SceneryTabInfo* GetSceneryTabInfoForGroup(const ObjectEntryIndex sceneryGroupIndex)
+    SceneryTabInfo* GetSceneryTabInfoForMisc()
     {
-        if (sceneryGroupIndex == OBJECT_ENTRY_INDEX_NULL)
+        for (auto& tabEntry : _tabEntries)
         {
-            return &_tabEntries[_tabEntries.size() - 1];
+            if (tabEntry.IsMisc())
+                return &tabEntry;
         }
 
+        return nullptr;
+    }
+
+    SceneryTabInfo* GetSceneryTabInfoForSearch()
```

Comment:
Likewise this always exists as the last tab.

Reply:
the search/all tab will be either the last tab or at `MaxTabsPerRow` - 1

Reply:
With how this was reworked recently it now always is the last tab. I've gone back and removed the unneeded for loops